source/common/stream_info/stream_info_impl.h

```diff
@@ -332,6 +338,7 @@ struct StreamInfoImpl : public StreamInfo {
   std::string route_name_;
   absl::optional<uint64_t> upstream_connection_id_;
   absl::optional<uint32_t> attempt_count_;
+  std::string& virtual_cluster_name_;
```

Comment:
This doesn't compile, right? By default there is no virtual_cluster_name and it gets set after construction?

Try making this `OptRef<const std::string>` and having accessors to this check virtual_cluster_name_.has_value().

Also -- it's worth double-checking that we are confident that the owner for this storage outlives the StreamInfo. If not you'll need to copy the string rather than referencing it, in which case you could just use `absl::optional<std::string>`.

Reply:
Changed it to `absl::optional<std::string>` and left a comment/TODO to investigate more for whether it could be a reference. Thanks for the suggestion.